sqlglot/dialects/clickhouse.py

```diff
@@ -328,3 +352,27 @@ def parameterizedagg_sql(self, expression: exp.Anonymous) -> str:
 
         def placeholder_sql(self, expression: exp.Placeholder) -> str:
             return f"{{{expression.name}: {self.sql(expression, 'kind')}}}"
+
+        def oncluster_sql(self, expression: exp.OnCluster) -> str:
```

Comment:
you need to ad this to the main generator or else we will error out because we can't handle this expression type

Reply:
When assembled programmatically?
Interesting. But isn't that's what `write=` is for?

Reply:
if you do write="" it will fail